scripts/Calibration/tofpd/group_calibration.py

```diff
@@ -126,39 +218,100 @@ def pdcalibration_groups(data_ws,
     :param PeakWidthPercent: PeakWidthPercent parameter of PDCalibration, default None
+    :param BadCalibThreshold: Threshold for relative difference between calibrated DIFC and engineering value.
     :return: tuple of DiffCal and Mask from CrossCorrelate combined with DiffCal from PDCalibration of grouped workspace
     """
 
+    CreateDetectorTable(data_ws, DetectorTableWorkspace="calib_table_bak")
+
     ApplyDiffCal(data_ws, CalibrationWorkspace=cc_diffcal)
     ConvertUnits(data_ws, Target='dSpacing', OutputWorkspace='_tmp_data_aligned')
     DiffractionFocussing('_tmp_data_aligned', GroupingWorkspace=group_ws, OutputWorkspace='_tmp_data_aligned')
 
     ConvertUnits('_tmp_data_aligned', Target='TOF', OutputWorkspace='_tmp_data_aligned')
 
-    PDCalibration(InputWorkspace='_tmp_data_aligned',
-                  TofBinning=TofBinning,
-                  PreviousCalibrationTable=previous_calibration,
-                  PeakFunction=PeakFunction,
-                  PeakPositions=PeakPositions,
-                  PeakWindow=PeakWindow,
-                  PeakWidthPercent=PeakWidthPercent,
-                  OutputCalibrationTable=f'{output_basename}_pd_diffcal',
-                  DiagnosticWorkspaces=f'{output_basename}_pd_diag')
+    instrument = data_ws.getInstrument().getName()
+
+    if instrument != 'POWGEN':
+        PDCalibration(InputWorkspace='_tmp_data_aligned',
+                      TofBinning=TofBinning,
+                      PreviousCalibrationTable=previous_calibration,
+                      PeakFunction=PeakFunction,
+                      PeakPositions=PeakPositions,
+                      PeakWindow=PeakWindow,
+                      PeakWidthPercent=PeakWidthPercent,
+                      OutputCalibrationTable=f'{output_basename}_pd_diffcal',
+                      DiagnosticWorkspaces=f'{output_basename}_pd_diag')
+    else:
+        PDCalibration(InputWorkspace='_tmp_data_aligned',
+                      TofBinning=TofBinning,
+                      PreviousCalibrationTable=previous_calibration,
+                      PeakFunction=PeakFunction,
+                      PeakPositions=PeakPositions,
+                      PeakWindow=PeakWindow,
+                      PeakWidthPercent=PeakWidthPercent,
+                      OutputCalibrationTable='PDCalib',
+                      DiagnosticWorkspaces='diag')
+        PDCalibration(InputWorkspace='_tmp_data_aligned',
+                      TofBinning=[TofBinning[0], TofBinning[1]/2, TofBinning[2]],
+                      PreviousCalibrationTable='PDCalib',
+                      PeakFunction=PeakFunction,
+                      PeakPositions=PeakPositions,
+                      PeakWindow=PeakWindow,
+                      PeakWidthPercent=PeakWidthPercent/2.0,
+                      OutputCalibrationTable='PDCalib',
+                      DiagnosticWorkspaces='diag')
+        PDCalibration(InputWorkspace='_tmp_data_aligned',
```

Comment:
Specific to POWGEN, I just refer to the calibration routine that is currently being used, which does contain three calls to `PDCalibration`. Given that it has been used for quite a while for POWGEN and turns out to be working robustly, I think we should stay consistent at this stage.

BTW, though I am not sure why we need three trials, definitely more than one trial is needed so that the next round of calibration can start from the previous result. Otherwise, a single run of `PDCalibration` starting from engineering position does not give reasonable result.